From ab2d1957b65aa5ba1bdd43284f6309a33c2fd97f Mon Sep 17 00:00:00 2001
From: "Thakur, Santosh Krx" <santosh.krx.thakur@intel.com>
Date: Thu, 4 Feb 2016 19:24:35 +0530
Subject: [PATCH 23/24] SoC: Intel: BYT: klockwork error in Audio Driver

This patch checks for NULL before dereferencing board_cfg.

Change-Id: If871d81500bc65fb37918380dd1981225ce4a523
Tracked-On: https://jira01.devtools.intel.com/browse/OAM-18599
Signed-off-by: Thakur, Santosh Krx <santosh.krx.thakur@intel.com>
Reviewed-on: https://android.intel.com:443/466546
---
 sound/soc/intel/board/byt_cr_dpcm_rt5640.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/sound/soc/intel/board/byt_cr_dpcm_rt5640.c b/sound/soc/intel/board/byt_cr_dpcm_rt5640.c
index 761e806..5f8882a 100644
--- a/sound/soc/intel/board/byt_cr_dpcm_rt5640.c
+++ b/sound/soc/intel/board/byt_cr_dpcm_rt5640.c
@@ -1031,6 +1031,11 @@ static int snd_byt_mc_probe(struct platform_device *pdev)
 
 	/* Get board-specific HW-settings */
 	drvdata->board_cfg = get_board_config(get_mc_link());
+	if (!drvdata->board_cfg) {
+		pr_err("%s: Failed to get board-specific HW-settings!\n",
+			__func__);
+		return -EFAULT;
+	}
 
 	/* Configure board-specific routes */
 	card = &snd_soc_card_byt_default;
-- 
1.9.1

